From 480a29fc39b1baaa32e2dd37a179483a0257f147 Mon Sep 17 00:00:00 2001
From: Lukas Bulwahn <lukas.bulwahn@gmail.com>
Date: Sat, 20 Apr 2019 04:01:41 +0200
Subject: [PATCH 26/79] trace: annotate t_start and t_stop

kernel/trace/trace.c:4291:1: warning: mutex 'trace_types_lock' is still held at the end of function [-Wthread-safety-analysis]
}
^
kernel/trace/trace.c:4284:2: note: mutex acquired here
        mutex_lock(&trace_types_lock);
        ^
kernel/trace/trace.c:4295:2: warning: releasing mutex 'trace_types_lock' that was not held [-Wthread-safety-analysis]
        mutex_unlock(&trace_types_lock);
        ^
---
 kernel/trace/trace.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/kernel/trace/trace.c b/kernel/trace/trace.c
index 6c24755655c7..50c70594130a 100644
--- a/kernel/trace/trace.c
+++ b/kernel/trace/trace.c
@@ -4275,7 +4275,7 @@ t_next(struct seq_file *m, void *v, loff_t *pos)
 	return t;
 }
 
-static void *t_start(struct seq_file *m, loff_t *pos)
+static void *t_start(struct seq_file *m, loff_t *pos) __attribute__((acquire_capability(trace_types_lock)))
 {
 	struct trace_array *tr = m->private;
 	struct tracer *t;
@@ -4290,7 +4290,7 @@ static void *t_start(struct seq_file *m, loff_t *pos)
 	return t;
 }
 
-static void t_stop(struct seq_file *m, void *p)
+static void t_stop(struct seq_file *m, void *p) __attribute__((release_capability(trace_types_lock)))
 {
 	mutex_unlock(&trace_types_lock);
 }
-- 
2.17.1

